File: tests/test_apps/csharp/ComprehensiveTest.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Newtonsoft.Json;
using Xunit;
using HtmlToMarkdown;

namespace HtmlToMarkdownTestApp;

#nullable enable

public record TestCase(string Name, string Html, string ExpectedMarkdown, Dictionary<string, object>? Options);

public class ComprehensiveTest
{
    private static List<TestCase> LoadFixtures(string filename)
    {
        // Get the directory of the test_apps folder (parent of csharp folder)
        var testAppsDir = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..", "..", "..");
        var fixturesPath = Path.Combine(testAppsDir, "fixtures", filename);
        var fullPath = Path.GetFullPath(fixturesPath);
        var json = File.ReadAllText(fullPath);
        return JsonConvert.DeserializeObject<List<TestCase>>(json)!;
    }

    public static IEnumerable<object[]> BasicHtmlData()
    {
        var fixtures = LoadFixtures("basic-html.json");
        return fixtures.Select(f => new object[] { f });
    }

    [Theory]
    [MemberData(nameof(BasicHtmlData))]
    public void TestBasicHtmlConversion(TestCase testCase)
    {
        var result = HtmlToMarkdownConverter.Convert(testCase.Html);
        Assert.Equal(testCase.ExpectedMarkdown.Trim(), result.Trim());
    }
}

File: tests/test_apps/go/comprehensive_test.go
package main

import (
	"encoding/json"
	"os"
	"path/filepath"
	"strings"
	"testing"
	htmltomarkdown "github.com/kreuzberg-dev/html-to-markdown/packages/go/v2/htmltomarkdown"
)

type TestCase struct {
	Name             string                 `json:"name"`
	HTML             string                 `json:"html"`
	ExpectedMarkdown string                 `json:"expectedMarkdown"`
	Options          map[string]interface{} `json:"options"`
}

func loadFixtures(t *testing.T, filename string) []TestCase {
	fixturePath := filepath.Join("..", "fixtures", filename)
	data, err := os.ReadFile(fixturePath)
	if err != nil {
		t.Fatalf("failed to read fixture: %v", err)
	}

	var cases []TestCase
	if err := json.Unmarshal(data, &cases); err != nil {
		t.Fatalf("failed to parse fixture: %v", err)
	}

	return cases
}

func TestBasicHTMLConversions(t *testing.T) {
	fixtures := loadFixtures(t, "basic-html.json")

	for _, tc := range fixtures {
		t.Run(tc.Name, func(t *testing.T) {
			result, err := htmltomarkdown.Convert(tc.HTML)
			if err != nil {
				t.Fatalf("conversion failed: %v", err)
			}

			if strings.TrimSpace(result) != strings.TrimSpace(tc.ExpectedMarkdown) {
				t.Errorf("expected: %q, got: %q", tc.ExpectedMarkdown, result)
			}
		})
	}
}
